src/constants.js

```diff
@@ -11,7 +9,8 @@ export const SC_STREAM_ATTR = 'data-styled-streamed';
 export const IS_BROWSER = typeof window !== 'undefined' && 'HTMLElement' in window;
 
 export const DISABLE_SPEEDY =
-  (typeof __DEV__ === 'boolean' && __DEV__) || process.env.NODE_ENV !== 'production';
+  (typeof process !== 'undefined' && !!process.env.SC_DISABLE_SPEEDY) ||
```

Comment:
@probablyup It looks like this was accidentally reverted with https://github.com/styled-components/styled-components/commit/0ecbb475fd370a83a91bd42591ebff395de10acd#diff-a7748843fbda5167d065231e3a2e7a1fR14